src/algorithms/base_algorithm.hpp

```diff
@@ -64,12 +64,13 @@ class BaseAlgorithm {
   std::shared_ptr<BaseMixing> mixing;
 
   // AUXILIARY TOOLS
+  virtual void add_datum_to_hierarchy(BaseHierarchy *hier, const int idx);
   //! Returns the values of an algo iteration as a Protobuf object
   bayesmix::MarginalState get_state_as_proto(unsigned int iter);
   //! Computes marginal contribution of a given iteration & cluster
   virtual Eigen::VectorXd lpdf_marginal_component(
       std::shared_ptr<BaseHierarchy> temp_hier,
-      const Eigen::MatrixXd &grid) = 0;
+      const std::vector<int> &idxs) = 0;
```

Comment:
As I already mentioned, whenever we want to evaluate a grid of points, we first have to build a fake hierarchy. Thus we might as well add another line which adds all appropriate data to that fake hierarchy. Not that it matters, since I will likely revert this changes